src/components/EnterpriseSubsidiesContext/data/hooks.js

```diff
@@ -2,24 +2,53 @@ import { useEffect, useState } from 'react';
 import { logError } from '@edx/frontend-platform/logging';
 import { getConfig } from '@edx/frontend-platform/config';
 import { camelCaseObject } from '@edx/frontend-platform/utils';
+import moment from 'moment';
 
 import EcommerceApiService from '../../../data/services/EcommerceApiService';
+import EnterpriseAccessApiService from '../../../data/services/EnterpriseAccessApiService';
 import LicenseManagerApiService from '../../../data/services/LicenseManagerAPIService';
+import SubsidyApiService from '../../../data/services/EnterpriseSubsidyApiService';
 
-export const useEnterpriseOffers = ({ enablePortalLearnerCreditManagementScreen }) => {
+export const useEnterpriseOffers = ({ enablePortalLearnerCreditManagementScreen, enterpriseId }) => {
   const [offers, setOffers] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [canManageLearnerCredit, setCanManageLearnerCredit] = useState(false);
 
   useEffect(() => {
     const fetchOffers = async () => {
       try {
-        const response = await EcommerceApiService.fetchEnterpriseOffers({
-          isCurrent: true,
-        });
-        const { results } = camelCaseObject(response.data);
-        setOffers(results);
+        const enterpriseAccessResponse = await EnterpriseAccessApiService.getSubsidyAccessPolicies(enterpriseId);
+        let { results } = camelCaseObject(enterpriseAccessResponse.data);
 
+        if (results.length === 0) {
+          // If there are no subsidy access policies in enterprise, fall back to the e-commerce API.
+          const response = await EcommerceApiService.fetchEnterpriseOffers({
```

Comment:
[curious/suggestion] Are these calls made sequentially? If so, we may want to make the calls in parallel, and only evaluate them once they are both loaded and then prioritize the `enterpriseAccessResponse`, for example: https://tanstack.com/query/v4/docs/react/guides/parallel-queries

Reply:
Great idea!